File: backend/App.Domain/Person.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using Base.Domain;

namespace App.Domain;

public class Person : BaseEntityIdMetadata
{
    [MaxLength(256)]
    public string FirstName { get; set; } = default!;
    
    [MaxLength(256)]
    public string LastName { get; set; } = default!;
    public int PersonalIdentificationNumber { get; set; }
    
    [JsonIgnore]
    public ICollection<ParticipantEvent>? ParticipantEvents { get; set; }
}

File: frontend/entities/Person.ts
import ParticipantEvent from "./ParticipantEvent";

export default class Person {
    id: string;
    firstName: string;
    lastName: string;
    personalIdentificationNumber: number;
    participantEvents: ParticipantEvent[];

    constructor(
        id: string,
        firstName: string,
        lastName: string,
        personalIdentificationNumber: number,
        participantEvents: ParticipantEvent[] 
    ) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.personalIdentificationNumber = personalIdentificationNumber;
        this.participantEvents = participantEvents 
    }
}
